custom_components/heatmiserneo/switch.py

```diff
@@ -9,15 +9,23 @@
 from datetime import time
 import logging
 
+import voluptuous as vol
 from homeassistant.const import EntityCategory
+from homeassistant.components.siren import SirenEntityFeature
```

Comment:
I'll let this slide for now in the interests of being pragmatic. However I have the feeling this is a bad anti pattern we don't want to rely on as it will end up causing a failure. I know that rolling up elements from other objects is valid but I strongly suggest maybe approaching the upstream maintainers and having duration added to the switch entity or perhaps creating a 'timer' entity we can consume safely. If they are happy and advise that this is valid behaviour, then I have no problem continuing. However when I re-visit this if it is not 'safe' I'll have to look at changing the behaviour.

Reply:
This is entirely optional. We can remove it for now if you prefer. The downside of removing it is that all switches from the integration (currently only two types; this and the standby switch) will appear in the entity selector. Then the user will need to make sure they select the right one for the hold to take effect.

I'll review some examples from HA core if they use this technique in any of the standard integrations (I doubt it). If I can't find any evidence of a similar hack, I'll remove it.

Reply:
I couldn't find any other examples doing something like this so I've removed it. Feels more inconvenient for the user, but less likely to break in the future